appium-provider/src/main/java/com/paypal/selion/appium/platform/grid/SeLionAppiumAndroidDriver.java

```diff
@@ -245,27 +246,40 @@ public void clearText(WebElement webElement) {
     @Override
     public void setText(WebElement webElement, String text) {
         logger.entering(webElement);
-        //As per the UI Object API doc a text field will be cleared before setting value
+        // As per the UI Object API doc a text field will be cleared before setting value
         webElement.clear();
         webElement.sendKeys(text);
         logger.exiting();
     }
 
     @Override
     public void swipe(int startx, int starty, int endx, int endy) {
-        //super.swipe(startx, starty, endx, endy, OPERATION_DURATION_MILLI_SECONDS );
+        swipe(startx, starty, endx, endy, OPERATION_DURATION_MILLI_SECONDS);
+    }
 
-        // On Appium Android we mimic swipe via one finger tap
-        this.tap(1, starty, endx, OPERATION_DURATION_MILLI_SECONDS);
+    public void swipe(int startx, int starty, int endx, int endy, int duration) {
+        new TouchAction(this).press(startx, starty).waitAction(Duration.ofMillis(duration)).moveTo(endx, endy).release().perform();
     }
```

Comment:
endx and endy - these are not absolute x,y value rather they represent the shift in x and y axis. So renaming them to shiftX and shiftY would make more sense

Also, consider a situation where you are swiping a carousel, most of the time you would be interested to go to a certain carousel content. To solve this problem if we can add a swipeCount argument then the author of the test can easily decide how many swipe is needed to go to the desired content, by default it will swipe only once.